lib/explorer/data_frame.ex

```diff
@@ -386,6 +386,22 @@ defmodule Explorer.DataFrame do
 
   ## Conversion
 
+  @doc """
+  Converts the dataframe to the lazy version of the current backend.
+
+  If already lazy, this is a noop.
+  """
+  @spec to_lazy(df :: DataFrame.t()) :: DataFrame.t()
+  def to_lazy(df), do: apply_impl(df, :to_lazy)
+
+  @doc """
+  This collects the lazy data frame into an eager one, computing the query.
+
+  If already eager, this is a noop.
+  """
+  @spec collect(df :: DataFrame.t()) :: {:ok, DataFrame.t()} | {:error, term()}
+  def collect(df), do: apply_impl(df, :collect)
```

Comment:
We should definitely have a bang variant then, so it flows nicer in the API. Although I wonder if we need the tuple version in the first place? I would think any failure while collecting would be a hard exception.

Reply:
I think you're right that it should be a hard exception. Good point.